svc-cloud-validation/src/main/kotlin/gov/cdc/dex/transport/service/ValidationController.kt

```diff
@@ -0,0 +1,114 @@
+package gov.cdc.dex.transport.service
+
+import gov.cdc.dex.cloud.storage.CloudStorage
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.MediaType
+import io.micronaut.http.annotation.Body
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Delete
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.PathVariable
+import io.micronaut.http.annotation.Post
+import io.micronaut.http.annotation.QueryValue
+import io.micronaut.http.multipart.CompletedFileUpload
+import org.slf4j.LoggerFactory
+import java.util.*
+//import gov.cdc.dex.hl7.Helper
+//import cdc.xlr.structurevalidator._
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+import java.io.BufferedReader
+import java.io.OutputStream
+
+import com.google.gson.*
+
+
+/**
+ *
+ *
+ * @Created - 6/23/23
+ * @Author USY6@cdc.gov
+ */
+@Controller("/validation")
+class ValidationController(private val cloudStorage: CloudStorage) {
+    private val log = LoggerFactory.getLogger(ValidationController::class.java.name)
+    val gson: Gson = GsonBuilder().serializeNulls().create()
+    val redactorUrl = System.getenv("REDACTOR_URL") + "/api/redactorReport"
+    val structureUrl = System.getenv("STRUCTURE_URL") + "/api/structure"
+    val validationUrl = System.getenv("MMG_VALIDATOR_URL") + "/api/validate-mmg"
+    val elrType = System.getenv("ELR_TYPE")
+
+    @Post(value = "/validation", consumes = [MediaType.APPLICATION_JSON, MediaType.TEXT_PLAIN])
+    fun uploadContentDefault(
+        @Body content: String,
+        @QueryValue fileContentType: String = MediaType.TEXT_PLAIN
+    ): HttpResponse<Any> {
+        val resultData = this.validateMessage(content);
+
+        return HttpResponse.ok(gson.toJson(resultData))
+    }
+
+    private fun validateMessage(hl7Content: String): String{
+        var reportData = ""
+
+        var redactedMessage = this.getContent(URL(this.redactorUrl), hl7Content, "ELR")
+
+        var structureReport = this.getContent(URL(this.structureUrl), redactedMessage, "CASE")
+
+        var validationReport = this.getContent(URL(this.validationUrl), redactedMessage, "CASE")
+
```

Comment:
The message type should be pulled from the HTTP header.  This is where you need to use the getMetadata function to retrieve these values:
![image](https://github.com/CDCgov/data-exchange-hl7/assets/111081583/d0daa54e-4b1c-446a-8bcc-90a1f8ff9ff9)

Reply:
Also you need to use a different namespace, not "transport.service" because that is something else